File: LATIHAN2DPBO2023/Java/Human.java
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 2
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//inisialisasi class Human
class Human
{
	//atribut
	private String nik;
	private String name;
	private String gender;

	//constructor
	public Human()
	{
		this.nik = "";
		this.name = "";
		this.gender = "";
	}

	//kumpulan method untuk class Human (setter dan getter)
	public void setNik(String nik)
	{
		this.nik = nik;
	}

	public String getNik()
	{
		return this.nik;
	}

	public void setName(String name)
	{
		this.name = name;
	}

	public String getName()
	{
		return this.name;
	}

	public void setGender(String gender)
	{
		this.gender = gender;
	}

	public String getGender()
	{
		return this.gender;
	}

	//pada java tidak memerlukan destructor karena sudah garbage collection
}

//inisialisasi class Sivitas yang merupakan child dari class Human
class Sivitas extends Human
{
	//atribut
	private String univ;
	private String email;

	//constructor
	public Sivitas()
	{
		this.univ = "";
		this.email = "";
	}

	//kumpulan method untuk class Sivitas (setter dan getter)
	public void setUniv(String univ)
	{
		this.univ = univ;
	}

	public String getUniv()
	{
		return this.univ;
	}

	public void setEmail(String email)
	{
		this.email = email;
	}

	public String getEmail()
	{
		return this.email;
	}
}

//inisialisasi class Mahasiswa yang merupakan child dari class Sivitas
class Mahasiswa extends Sivitas
{
	//atribut
	private String nim;
	private String faculty;
	private String major;

	//constructor
	public Mahasiswa()
	{
		this.nim = "";
		this.faculty = "";
		this.major = "";
	}

	//kumpulan method untuk class Mahasiswa (setter dan getter)
	public void setNim(String nim)
	{
		this.nim = nim;
	}

	public String getNim()
	{
		return this.nim;
	}

	public void setFaculty(String faculty)
	{
		this.faculty = faculty;
	}

	public String getFaculty()
	{
		return this.faculty;
	}

	public void setMajor(String major)
	{
		this.major = major;
	}

	public String getMajor()
	{
		return this.major;
	}
}

File: LATIHAN2DPBO2023/Python/Human.py
# Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 2
# dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
# untuk keberkahanNya maka saya tidak melakukan kecurangan 
# seperti yang telah dispesifikasikan. Aamiin.

# note: untuk penjelasan desain algoritma (inheritance, dll) ada di readme github

# inisialisasi class Human (sebagai base class / parent class utama)
class Human:
	# atribut
	__nik = ""
	__name = ""
	__gender = ""

	# constructor
	def __init__(self, nik = "", name = "", gender = ""):
		self.__nik = nik
		self.__name = name
		self.__gender = gender

	# kumpulan method setter dan getter untuk tiap atribut
	def setNik(self, nik):
		self.__nik = nik

	def getNik(self):
		return self.__nik

	def setName(self, name):
		self.__name = name

	def getName(self):
		return self.__name

	def setGender(self, gender):
		self.__gender = gender

	def getGender(self):
		return self.__gender

	#python tidak memerlukan destructor karena sudah garbage collection

# inisialisasi class Sivitas sebagai child dari class Human
class Sivitas(Human):
	# atribut
	__univ = ""
	__email = ""

	# constructor
	def __init__(self, univ = "", email = ""):
		self.__univ = univ
		self.__email = email

	# kumpulan method setter dan getter untuk tiap atribut
	def setUniv(self, univ):
		self.__univ = univ

	def getUniv(self):
		return self.__univ

	def setEmail(self, email):
		self.__email = email

	def getEmail(self):
		return self.__email

# inisialisasi class Mahasiswa sebagai child dari class Sivitas (juga "cucu" dari class Human)
class Mahasiswa(Sivitas):
	# atribut
	__nim = ""
	__faculty = ""
	__major = ""

	# constructor
	def __init__(self, nim = "", faculty = "", major = ""):
		self.__nim = nim
		self.__faculty =faculty
		self.__major =major

	# kumpulan method setter dan getter untuk tiap atribut
	def setNim(self, nim):
		self.__nim = nim

	def getNim(self):
		return self.__nim

	def setFaculty(self, faculty):
		self.__faculty = faculty

	def getFaculty(self):
		return self.__faculty

	def setMajor(self, major):
		self.__major = major

	def getMajor(self):
		return self.__major